lib/semian/net_http.rb

```diff
@@ -0,0 +1,94 @@
+require 'semian'
+require 'semian/adapter'
+
+module Net
+  ProtocolError.include(::Semian::AdapterError)
+
+  class SemianError < ::Net::ProtocolError
+    def initialize(semian_identifier, *args)
+      super(*args)
+      @semian_identifier = semian_identifier
+    end
+  end
+
+  ResourceBusyError = Class.new(SemianError)
+  CircuitOpenError = Class.new(SemianError)
+end
+
+module Semian
+  module NetHTTP
+    include Semian::Adapter
+
+    ResourceBusyError = ::Net::ResourceBusyError
+    CircuitOpenError = ::Net::CircuitOpenError
+
+    def semian_identifier
+      @configuration ||= Semian::NetHTTP.retrieve_semian_configuration_by_host_port(address, port)
```

Comment:
Meh, I'll see. If I do the `:name` thing, then I'll want to cache the `Hash` with the `:name` key-value pair removed. So it might not work like ^ at all.